spirv-builder/src/test/mod.rs

```diff
@@ -0,0 +1,69 @@
+mod basic;
+
+use lazy_static::lazy_static;
+use std::error::Error;
+use std::path::{Path, PathBuf};
+use std::sync::Mutex;
+
+// Tests need to run serially, since they write project files to disk and whatnot. We don't want to
+// create a new temp dir for every test, though, since then every test would need to build libcore.
+// We could require the user to pass --thread-count 1 to cargo test, but that affects other tests.
+// So make a global mutex wrapping every test.
+lazy_static! {
+    static ref GLOBAL_MUTEX: Mutex<()> = Mutex::new(());
+}
+
+static CARGO_TOML: &str = r#"[package]
+name = "test-project"
+version = "0.1.0"
+authors = ["Embark <opensource@embark-studios.com>"]
+edition = "2018"
+
+[lib]
+crate-type = ["dylib"]
+
+[dependencies]
+spirv-std = { path = "../../spirv-std" }
+
+[workspace]
+"#;
+
+static SRC_PREFIX: &str = r#"#![no_std]
+#![feature(lang_items, register_attr)]
+#![register_attr(spirv)]
+use core::panic::PanicInfo;
+use spirv_std::*;
+#[panic_handler]
+fn panic(_: &PanicInfo) -> ! {
+    loop {}
+}
+#[lang = "eh_personality"]
+extern "C" fn rust_eh_personality() {}
```

Comment:
You should be able to avoid this if you pass `-Cpanic=abort` to rustc.

Reply:
There's a whole complex backstory here, I would explain, but, currently dealing with some horrible garbage (see _fltused windows CI failure, which is the latest item on the heap of garbage), so, bleh :(